hook!-full-game/src/Player/States/Wall.gd

```diff
@@ -0,0 +1,39 @@
+extends State
+
+const WALL_SLIDE_SPEED: = 100.0
+const WALL_PULL_STRENGTH: = 300.0
+
+var _jump_strength: = 1000.0
+var _velocity: = Vector2.ZERO
+var _wall_direction: = -1
+
+func enter(msg: Dictionary = {}) -> void:
+	_wall_direction = msg.direction
+
+
+func physics_process(delta: float) -> void:
+	_velocity.y += WALL_SLIDE_SPEED * delta
+	_velocity.x += (WALL_PULL_STRENGTH * -_wall_direction) * delta
```

Comment:
As you constantly use the velocity that move_and_slide gives you, this line doesn't have any impact.